content/docs/engine-dev/tools/tiaf/maintaining-tiaf/debugging-tiaf.md

```diff
@@ -77,47 +87,182 @@ In the log above, you can extract the following information about the AR run and
+
+The script configuration file used for this step is as follows:
+
+```json
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "Python: Current File",
+            "type": "python",
+            "request": "launch",
+            "program": "${file}",
+            "console": "integratedTerminal",
+            "justMyCode": true,
+            "args": [
+                "--config", "C:/dev/o3de/build/windows_vs2019/bin/TestImpactFramework/debug/Persistent/tiaf.json", // Replace with your build path to tiaf.json
+                "--src-branch", "development", // src and dst must be same in order to write out historic data
+                "--dst-branch", "development", // Note: the src and dst branches can be any name and do not need to refer to real, existing branches
+                "--commit", "4b65521188a2a0beb399eca7b692216d2fe3d209",
+                "--mars-index-prefix", "o3de-tiaf",
+                "--build-number", "001", // The build number can be anything
+                "--suites", "smoke", "main", // The suites must match those used in AR
+                "--label-excludes", "REQUIRES_gpu", // The label exclude filters must match those used in AR
+                "--test-failure-policy", "continue",
+                "--target-output", "stdout",
+                "--runtime-type", "native" // change to "python" for Python TIAF
+            ]
+        }
+    ]
+}
+```
+
+When the script has completed execution you will find the same historic data used in AR at the location `<build_path>/bin/TestImpactFramework/debug/Persistent/Native/historic/main-smoke`.
+
+{{< tip >}}
+Subsequent runs of CMake will delete the contents of this directory. If you wish to place the historic data in another location to preserve it between runs of CMake, copy the `tiaf.json` file from your build location to somewhere convenient and modify the `root` path in the `historic` section of the appropriate runtime of this file to this new location. Now you can change the path for `config` argument in the `tiaf_driver.py` script to that of the `tiaf.json` configuration file you modified.
+{{< /tip >}}
 
-  ```
-  --config=<path to your config file> --src-branch=stabilization/2305 --dst-branch= --commit=59e24a62859ce8a932e338f36432e3228621c1ec --mars-index-prefix=o3de-tiaf --build-number=<pick any number> --suites smoke main --label-excludes REQUIRES_gpu --test-failure-policy=continue --runtime-type=python --testrunner=live --target-output=stdout
-  ```
+### Step 2: Build and run the commit failing in AR
```

Comment:
@jonawals I noticed that my src branch is `development` in step 1, and `PR-15830` in step 2. Is the reason for this so I can create that diff between the historic data? It would be helpful if you can explain the reasoning here or clarify the comments in the code snippets better.

Reply:
You will need the coverage data generated from Step 1 in order to faithfully produce the state of AR in Step 2.